server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java

```diff
@@ -280,6 +280,21 @@ private static Long parseDateTimeAsLong(@Nullable String dateAsString) {
     return date.getTime();
   }
 
+  @CheckForNull
+  private static Long parseFromDate(@Nullable String dateAsString) {
+    if (dateAsString == null) {
+      return null;
+    }
+
+    Date date = parseDateTimeQuietly(dateAsString);
+    if (date == null) {
+      date = parseDateQuietly(dateAsString);
+      checkRequest(date != null, "Date '%s' cannot be parsed as either a date or date+time", dateAsString);
+    }
+
+    return date.getTime();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'date' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)